### DISCUSSION TITLE
Middleware rewrites

### DISCUSSION THREAD

**Discussion by ZeldOcarina** [vercel/community#5275] [Feature Question]
### Question

I've got from SEO a huge list of rewrites to be made for a website so I've got a huge matcher:

![image](https://github.com/vercel/community/assets/42982095/daf70573-6e1c-452d-8360-ba6658a30952)

This is just abt 10%,

How can I accomodate this? The build would outright fail with:

```
Error: Builder returned invalid routes: should NOT be longer than 4096 characters
```

Thanks for your help.

I am working with Astro.

**Comment by molebox** [comment]
Hey @ZeldOcarina You could try using [Edge Config](https://vercel.com/docs/storage/edge-config) to store the redirects, and then read from that in your Edge Middleware. There is a. [limits to read and writes on Edge Config](https://vercel.com/docs/storage/edge-config/edge-config-limits#) though, How many redirect paths do you have?

**Reply by ZeldOcarina** [reply]
Hey @molebox thanks a lot for your help. Right now it's only 43 but it will increase for sure!

**Reply by z4nr34l** [reply]
The best solution, would be using edge config for storing this list just like it was said, but read them and redirect/rewrite in next.config.js or just simple put all of them in rewrites inside vercel.json file.

Related resources that will help you:

https://vercel.com/docs/projects/project-configuration#redirects

https://nextjs.org/docs/app/api-reference/next-config-js/redirects

**Reply by ZeldOcarina** [reply]
I am using Astro, not Next, here it says to avoid using vercel.json:

[Rewrites](https://vercel.com/docs/frameworks/astro#rewrites)

Also, I tried building Astro config the config on the fly at build time to implement the behavior but it did not work..
